ReSharper.FSharp/src/FSharp.Psi.Features/src/Util/FSharpResolveUtil.fs

```diff
@@ -81,3 +82,13 @@ let mayShadowPartially (newExpr: ITreeNode) (data: ElementProblemAnalyzerData) (
     |> Seq.collect (fun element -> opens.GetValuesSafe(element.ShortName))
     |> Seq.toArray
     |> OpenScope.inAnyScope newExpr
+
+let isPredefinedFunctionShadowed (context: ITreeNode) name opName =
+    let checkerService = context.GetContainingFile().As<IFSharpFile>().CheckerService
+    match checkerService.ResolveNameAtLocation(context, [name], opName) with
+    | Some symbolUse ->
+        let symbol = symbolUse.Symbol
+        match predefinedFunctionTypes.TryGetValue(symbol.DisplayName) with
+        | true, fullName -> (fullName.FullName + "." + symbol.DisplayName) <> symbol.FullName
+        | _ -> false
+    | None -> false
```

Comment:
It's an interesting case. I can think of only two situations where a builtin function like `id` won't resolve to any symbol:
* missing FSharp.Core reference (should never happen in normal conditions, since FCS would fallback to the bundled one, despite bad fsproj/IDE configuration, but it _might_ fail to find one)
* editing FSharp.Core in a context prior to the function definition.

It's indeed not being shadowed but at the same time it's not even available in the context and we should not add a reference to it in the quick fix. I'd prefer to be more defensive here and to not allow the quick fix (moreover, I think we should not produce a warning here at all). Could you change the function to something like `resolvesToPredefinedFunction` and update the quick fix usage accordingly?